lib/raty.js

```diff
@@ -0,0 +1,784 @@
+    var decimal = this._getDecimal(score, 1);
+
+    if (decimal !== 0) {
+      decimal = decimal > 5 ? 1 : 0.5;
+    }
+
+    return integer + decimal;
+  }
+
+  // TODO: model spec
+  _roundStars(score, evt) {
+    var name = this._starName(score, evt);
+    if (name) {
+      var icon = this.opt[name];
+      var star = this.stars[Math.ceil(score) - 1];
+      this._setIcon(star, icon);
+    } // Full down: [x.00 .. x.25]
+  }
+
+  // TODO: model spec
+  _setIcon(star, icon) {
+    star[this.opt.starType === 'img' ? 'src' : 'className'] = this.opt.path + icon;
+  }
+
+  _setPath() {
+    this.opt.path = this.opt.path || '';
+
+    if (this.opt.path && this.opt.path.slice(-1)[0] !== '/') {
+      this.opt.path += '/';
+    }
+  }
+
+  // TODO: model spec
+  _setTarget(target, score) {
+    if (score) {
+      score = this.opt.targetFormat.toString().replace('{score}', score);
+    }
+    if (target instanceof HTMLInputElement || target instanceof HTMLSelectElement) {
+      target.value = score;
+    } else {
+      target.textContent = score;
+    }
+  }
+
+  // TODO: model spec
+  _setTitle(score, evt) {
+    if (score) {
+      var integer = parseInt(Math.ceil(score), 10);
+      var star = this.stars[integer - 1];
+
+      star.title = this._getHint(score, evt);
+    }
+  }
+
+  _starName(score, evt) {
+    var decimal = +(score % 1).toFixed(2);
+
+    if (evt || this.isMove) {
+      return decimal > 0.5 ? 'starOn' : 'starHalf';
+    }
+
+    if (decimal <= this.opt.round.down) {
+      // Down: [x.00 ... x.25]
+      return;
+    }
+
+    if (this.opt.halfShow && decimal < this.opt.round.up) {
+      // Half: [x.26 ... x.75]
+      return 'starHalf';
+    }
+
+    if (decimal < this.opt.round.full) {
+      // Off: [x.26 .. x.6]
+      return 'starOff';
+    }
+
+    return 'starOn'; // Up: [x.26 ...] || [x.6 ...]
+  }
+  // TODO: model spec
+  _target(score, evt) {
+    if (this.opt.target) {
+      var target = document.querySelector(this.opt.target);
+
+      if (!target) {
+        this._error('Target selector invalid or missing!');
+      }
+
+      var mouseover = evt && evt.type === 'mouseover';
+
+      if (score === undefined) {
+        score = this.opt.targetText;
+      } else if (score === null) {
+        score = mouseover ? this.opt.cancelHint : this.opt.targetText;
+      } else {
+        if (this.opt.targetType === 'hint') {
+          score = this._getHint(score, evt);
+        } else if (this.opt.precision) {
+          score = parseFloat(score).toFixed(1);
+        }
+
+        var mousemove = evt && evt.type === 'mousemove';
+
+        if (!mouseover && !mousemove && !this.opt.targetKeep) {
+          score = this.opt.targetText;
+        }
+      }
+
+      this._setTarget(target, score);
+    }
+  }
+
+  // TODO: model spec
+  _turnOn(i, score) {
+    return this.opt.single ? i === score : i <= score;
+  }
+
+  // TODO: model spec
+  _unlock() {
+    this.self.style.pointerEvents = 'auto';
```

Comment:
Here we were using `pointer`.

Reply:
pointerEvent is better than pointer 

[Please check this] (https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events)